collector/snapshots.go

```diff
@@ -276,20 +283,28 @@ func (s *Snapshots) Collect(ch chan<- prometheus.Metric) {
 		ch <- s.jsonParseFailures
 	}()
 
-	// indices
-	snapshotsStatsResp, err := s.fetchAndDecodeSnapshotsStats()
-	if err != nil {
+	if s.updateInterval == 0 {
+		_ = level.Debug(s.logger).Log("msg", "getting snapshots metrics", "block", "true")
+		s.lastResponse, s.lastError = s.fetchAndDecodeSnapshotsStats()
+	} else if s.lastFetchAt.Add(s.updateInterval).Before(time.Now()) {
+		go func() {
+			_ = level.Debug(s.logger).Log("msg", "getting snapshots metrics", "block", "false")
+			s.lastFetchAt = time.Now()
```

Comment:
[According to the docs](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus?tab=doc#Collector), the `Collect()` method may be called concurrently so it should be  concurrency safe which I do not believe these additions adhere to. `lastFetchAt`, `lastResponse`, and `lastErrror` can all change during the lifetime of this method call. I think there needs to be some sort of mutex to handle the concurrency or there may be a way to refactor in order to avoid the need for a mutex.

Reply:
done, thanks for the feedback :)